app/src/swap.c

```diff
@@ -0,0 +1,252 @@
+    if (params == NULL) {
+        return;
+    }
+    params->result = 0;
+    uint8_t buffer[100] = {0};
+    uint16_t replyLen = 0;
+
+    // address parameters have the following structure
+    // address kind (1 byte) | path length (1 byte) | bip44 path (4 * pathLength bytes)
+    const key_kind_e add_kind = (key_kind_e) *params->address_parameters;
+    zxerr_t err = extractHDPath((uint8_t*) params->address_parameters + 2, (uint8_t) params->address_parameters_length - 2);
+    if (params->address_to_check == 0 || err != zxerr_ok) {
+        return;
+    }
+
+    err = crypto_fillAddress(add_kind, buffer, sizeof(buffer), &replyLen);
+
+    if (err != zxerr_ok || replyLen <= PK_LEN_25519) {
+        MEMZERO(buffer, sizeof(buffer));
+        return;
+    }
+
+    const uint8_t *address = buffer + PK_LEN_25519;
+    const uint8_t addressLen = replyLen - PK_LEN_25519;
+    const uint8_t addressToCheckLen = strlen(params->address_to_check);
+
+    if (addressLen == addressToCheckLen &&
+        memcmp(address, params->address_to_check, addressLen) == 0) {
+        params->result = 1;
+    }
+}
+
+void handle_get_printable_amount( get_printable_amount_parameters_t* params) {
+    if (params == NULL) {
+        return;
+    }
+    MEMZERO(params->printable_amount, sizeof(params->printable_amount));
+
+    // Convert to BCD
+    bignumBigEndian_to_bcd((uint8_t*) params->printable_amount, sizeof(params->printable_amount),
+                           params->amount, params->amount_length);
+
+    // Format number.
+    if (!intstr_to_fpstr_inplace(params->printable_amount, sizeof(params->printable_amount), COIN_AMOUNT_DECIMAL_PLACES)) {
+        return;
+    }
+
+    // Add ticker prefix.
+    if (z_str3join(params->printable_amount, sizeof(params->printable_amount), COIN_TICKER, "") != zxerr_ok) {
+        return;
+    }
+
+    number_inplace_trimming(params->printable_amount, 1);
+}
+
+bool copy_transaction_parameters(create_transaction_parameters_t* sign_transaction_params) {
+    if (sign_transaction_params == NULL) {
+        return false;
+    }
+    MEMZERO(&G_swap_state, sizeof(G_swap_state));
+
+    strncpy(G_swap_state.destination_address,
+            sign_transaction_params->destination_address,
+            sizeof(G_swap_state.destination_address) - 1);
+
+    if ((G_swap_state.destination_address[sizeof(G_swap_state.destination_address) - 1] != '\0') ||
+        (sign_transaction_params->amount_length > 8) ||
+        (sign_transaction_params->fee_amount_length > 8)) {
+        return false;
+    }
+
+    G_swap_state.amount_length = sign_transaction_params->amount_length;
+
+    // store amount as big endian in 8 bytes, so the passed data should be alligned to right
+    // input {0xEE, 0x00, 0xFF} should be stored like {0x00, 0x00, 0x00, 0x00, 0x00, 0xEE, 0x00, 0xFF}
+    memcpy(G_swap_state.amount + 8 - sign_transaction_params->amount_length, sign_transaction_params->amount, sign_transaction_params->amount_length);
```

Comment:
@carlosala You need to simply store the amount, no need for the alignment : 

 `memcpy(G_swap_state.amount, sign_transaction_params->amount, sign_transaction_params->amount_length);`